From 0599a9d0acdf06d8e68885081ffb7da6076ebaef Mon Sep 17 00:00:00 2001
From: Peter Ujfalusi <peter.ujfalusi@ti.com>
Date: Fri, 15 Nov 2013 14:02:42 +0200
Subject: [PATCH 1017/1115] ASoC: davinci-evm: Fix evm_get_bclk() regarding to
 channels and move it up in the file

We should take the channels from the params and not pass a static '2' assuming
that the stream is stereo.
Move it up in the file so other functions can call it if they need this.

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@ti.com>
(cherry picked from commit 8c7b07a3bdf641763699242d0b401998566a6db2)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 sound/soc/davinci/davinci-evm.c |   20 ++++++++++----------
 1 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/sound/soc/davinci/davinci-evm.c b/sound/soc/davinci/davinci-evm.c
index 7fa5232..b5d45c3 100644
--- a/sound/soc/davinci/davinci-evm.c
+++ b/sound/soc/davinci/davinci-evm.c
@@ -42,6 +42,15 @@ struct snd_soc_card_drvdata_davinci {
    to be changed. */
 #define TDA998X_SAMPLE_FORMAT SNDRV_PCM_FORMAT_S32_LE
 
+static unsigned int evm_get_bclk(struct snd_pcm_hw_params *params)
+{
+	int sample_size = snd_pcm_format_width(params_format(params));
+	int rate = params_rate(params);
+	int channels = params_channels(params);
+
+	return sample_size * channels * rate;
+}
+
 static int evm_startup(struct snd_pcm_substream *substream)
 {
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
@@ -115,15 +124,6 @@ static int evm_tda998x_startup(struct snd_pcm_substream *substream)
 	return evm_startup(substream);
 }
 
-static unsigned int evm_get_bclk(struct snd_pcm_hw_params *params,
-				 int channels)
-{
-	int sample_size = snd_pcm_format_width(params_format(params));
-	int rate = params_rate(params);
-
-	return sample_size * channels * rate;
-}
-
 static int evm_tda998x_hw_params(struct snd_pcm_substream *substream,
 				 struct snd_pcm_hw_params *params)
 {
@@ -132,7 +132,7 @@ static int evm_tda998x_hw_params(struct snd_pcm_substream *substream,
 	struct snd_soc_codec *codec = rtd->codec;
 	struct snd_soc_card *soc_card = codec->card;
 	struct platform_device *pdev = to_platform_device(soc_card->dev);
-	unsigned int bclk_freq = evm_get_bclk(params, 2);
+	unsigned int bclk_freq = evm_get_bclk(params);
 	unsigned sysclk = ((struct snd_soc_card_drvdata_davinci *)
 			   snd_soc_card_get_drvdata(soc_card))->sysclk;
 	int ret;
-- 
1.7.5.4

